File: src/main/java/de/florianschmitt/service/util/TransactionHook.java
package de.florianschmitt.service.util;

import org.springframework.transaction.support.TransactionSynchronizationAdapter;
import org.springframework.transaction.support.TransactionSynchronizationManager;

public class TransactionHook {
    public static void afterCommitSuccess(Runnable work) {
        TransactionSynchronizationManager.registerSynchronization(new TransactionSynchronizationAdapter() {

            @Override
            public void afterCommit() {
                work.run();
            }
        });
    }
}

File: src/main/kotlin/de/florianschmitt/service/util/TransactionHook.kt
package de.florianschmitt.service.util

import org.springframework.transaction.support.TransactionSynchronizationAdapter
import org.springframework.transaction.support.TransactionSynchronizationManager

object TransactionHook {
    fun afterCommitSuccess(work: () -> Unit) {
        TransactionSynchronizationManager.registerSynchronization(object : TransactionSynchronizationAdapter() {

            override fun afterCommit() {
                work.invoke()
            }
        })
    }
}
